### DISCUSSION TITLE
Which Input is recommanded on step re-annotation, all or filtered?

### DISCUSSION THREAD

**Discussion by zhouzhendiao** [tuonglab/dandelion#278] [Q&A] [answered]
In tutorial [preprocessing chapter](https://sc-dandelion.readthedocs.io/en/latest/notebooks/Q1-singularity-preprocessing.html), `all_contig_annotations.csv` is used as input. However, in [load data for analysis](https://sc-dandelion.readthedocs.io/en/latest/notebooks/Q2-object-prep.html), input changed to `filtered_contig_annotations.csv`. I wonder which input is recommaned, raw one or filtered one? Does dandelion preprocessing pipeline filter the annotations of contig?

**Comment by zktuong** [comment, marked as answer]
hi @zhouzhendiao - i apologise for the confusion. In practice i think if you are working on mature T/B cells, then `filtered_contig_annotations.csv` would probably be alright. We started using `all_contig_annotations.csv` because we were working on the developing datasets in the past couple of papers by Suo et al. I would suspect that you would probably end up with similar recovery with either formats. The trade off for me is about speed vs how thorough you want your analysis - the all_contig file has a lot of contigs that dandelion will have to run through but will process everything but filter_contig gives you a quick pass at the data. Hope this helps

**Reply by zhouzhendiao** [reply]
OK, I got it. Thanks!
